modules/workflow-workflowoperation/src/main/java/org/opencastproject/workflow/handler/workflow/ConfigureByDublinCoreTermWOH.java

```diff
@@ -96,29 +100,48 @@ public WorkflowOperationResult start(WorkflowInstance workflowInstance, JobConte
 
     String configuredCatalog = StringUtils.trimToEmpty(currentOperation.getConfiguration(DCCATALOG_PROPERTY));
     String configuredDCTerm = StringUtils.trimToEmpty(currentOperation.getConfiguration(DCTERM_PROPERTY));
+    String configuredExternalCatalog = StringUtils.trimToEmpty(currentOperation.getConfiguration(EXTERNAL_CATALOG_PROPERTY));
     String configuredDefaultValue = StringUtils.trimToNull(currentOperation.getConfiguration(DEFAULT_VALUE_PROPERTY));
     String configuredMatchValue = StringUtils.trimToEmpty(currentOperation.getConfiguration(MATCH_VALUE_PROPERTY));
 
+    // Set Default
+    EName dcterm = null;
+    if (("").equals(configuredExternalCatalog)) {
+      configuredExternalCatalog = "dublincore";
+      dcterm = new EName(TERMS_NS_URI, configuredDCTerm);
+    }
     // Find Catalog
     Catalog[] catalogs = mediaPackage
-            .getCatalogs(new MediaPackageElementFlavor("dublincore", StringUtils.lowerCase(configuredCatalog)));
+            .getCatalogs(new MediaPackageElementFlavor(configuredExternalCatalog, StringUtils.lowerCase(configuredCatalog)));
 
     if (catalogs != null && catalogs.length > 0) {
       Boolean foundValue = false;
-      EName dcterm = new EName(TERMS_NS_URI, configuredDCTerm);
 
       // Find DCTerm
+      List<DublinCoreValue> values = new LinkedList<>();
       for (Catalog catalog : catalogs) {
         DublinCoreCatalog dc = DublinCoreUtil.loadDublinCore(workspace, catalog);
-        // Match Value
-        List<DublinCoreValue> values = dc.get(dcterm);
-        if (values.isEmpty()) {
-          // Use default
-          if (configuredDefaultValue != null) {
-            foundValue = configuredDefaultValue.equals(configuredMatchValue);
+          for (Map.Entry<EName, List<DublinCoreValue>> entry : dc.getValues().entrySet()) {
+            if (dcterm != null && dcterm.equals(entry)) {
+              values.addAll(entry.getValue());
+            } else if (entry.getKey().getLocalName().equals(configuredDCTerm)) {
+              values.addAll(entry.getValue());
+            }
           }
-        } else {
-          foundValue = values.contains(DublinCoreValue.mk(configuredMatchValue));
+      }
+      // Match Value
+      if (values.isEmpty()) {
+        // Use default
+        if (configuredDefaultValue != null) {
+          // check for direct character match or regex match
+          foundValue = configuredDefaultValue.equals(configuredMatchValue)
+                  || configuredDefaultValue.matches(configuredMatchValue);
```

Comment:
`.matches(…)` should always be true if `.equals(…)` if true, won't it? That means that `.equals(…)` is just superfluous.

Reply:
First I check if the strings are identical (current default case) second I check if "configuredMatchValue" might be a regex that could match. If I only would have an match, if the regex always includes the identical value also.